model_evaluation/Predict bias.ipynb

```diff
@@ -0,0 +1,370 @@
+      "('signal_early_stopping/STOP', [])\n",
+      "('title/beta1_power', [])\n",
+      "('title/beta2_power', [])\n"
+     ]
+    }
+   ],
+   "source": [
+    "var_list = checkpoint_utils.list_variables(MODEL_DIR_OCCUPATION)\n",
+    "for v in var_list:\n",
+    "    print(v)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 99,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "kernel_occupation = np.transpose(checkpoint_utils.load_variable(MODEL_DIR_OCCUPATION, 'dense_2/kernel'))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 100,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "('beta1_power', [])\n",
+      "('beta2_power', [])\n",
+      "('dense/bias', [1])\n",
+      "('dense/kernel', [256, 1])\n",
+      "('dense_1/bias', [128])\n",
+      "('dense_1/kernel', [256, 128])\n",
+      "('embeddings', [400002, 100])\n",
+      "('final_layer/bias', [33])\n",
+      "('final_layer/bias/Adam', [33])\n",
+      "('final_layer/bias/Adam_1', [33])\n",
+      "('final_layer/kernel', [128, 33])\n",
+      "('final_layer/kernel/Adam', [128, 33])\n",
+      "('final_layer/kernel/Adam_1', [128, 33])\n",
+      "('global_step', [])\n",
+      "('rnn/multi_rnn_cell/cell_0/gru_cell/candidate/bias', [256])\n",
+      "('rnn/multi_rnn_cell/cell_0/gru_cell/candidate/kernel', [356, 256])\n",
+      "('rnn/multi_rnn_cell/cell_0/gru_cell/gates/bias', [512])\n",
+      "('rnn/multi_rnn_cell/cell_0/gru_cell/gates/kernel', [356, 512])\n",
+      "('signal_early_stopping/STOP', [])\n"
+     ]
+    }
+   ],
+   "source": [
+    "var_list = checkpoint_utils.list_variables(MODEL_DIR_GENDER)\n",
+    "for v in var_list:\n",
+    "    print(v)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 101,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "kernel_gender = np.transpose(checkpoint_utils.load_variable(MODEL_DIR_GENDER, 'final_layer/kernel'))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Compute cosinus."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 102,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "TITLE_LABELS = [\n",
+    "    'accountant', 'acupuncturist', 'architect', 'attorney', 'chiropractor', 'comedian', 'composer', 'dentist',\n",
+    "    'dietitian', 'dj', 'filmmaker', 'interior_designer', 'journalist', 'landscape_architect', 'magician',\n",
+    "    'massage_therapist', 'model', 'nurse', 'painter', 'paralegal', 'pastor', 'personal_trainer',\n",
+    "    'photographer', 'physician', 'poet', 'professor', 'psychologist', 'rapper',\n",
+    "    'real_estate_broker', 'software_engineer', 'surgeon', 'teacher', 'yoga_teacher']"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 103,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(33, 128)"
```

Comment:
Is this the shape we expect for kernel_gender? Why not (2, 128) for just the two genders (male/female)?

Reply:
This is a great question.
This model had actually a bug: I had put n_classes=33 even if the only classes were {0,1}. Do you think that it might affect this analysis? I can retrain a model if needed.

Also, related to what we mentioned yesterday, we could discuss (2,128) vs (1,128). I was thinking that (2,128) makes our analysis a little harder, but is more generalizable to non-binary identity groups.
Let me know what you think

Reply:
I think retraining with n_classes = 1 makes sense here. Having a unique "gender indicator" vector (basically a gender direction) will make the subsequent analysis easier. For extending to non-binary identity groups, this may still be the best path as we want an indicator direction for each member of the group (and it's not clear that the mean-subtraction techniques you use in the other notebook are better).

Reply:
Sounds good. Code has been modified and new models will incorporate this change